### PR TITLE
Xdr context bug fix

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25271

## Description
There were actually 2 bugs that lead to this issue:
1. The context entry DT had a typo (was` val.endpoint_id == **val**.endpoinid` instead of ` val.endpoint_id == obj.endpoinid`) all such typos were fixed.
2. The get-endpoints command had 2 different flows where for one the ID was outputted as `endpoint_id` and for the second was outputted as `agent_id`. Added the `endpoint_id` key to the second scenario as well (did not remove the `agent_id` key to support BC)

## Screenshots
**Before the fix:** 
Test_XDR_Playbook_Wed_Jun_03_2020 (1)

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests

### CODE REVIEW

File: Packs/CortexXDR/pack_metadata.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "Palo Alto Networks Cortex XDR - Investigation and Response",
     "description": "Cortex XDR is the world's first detection and response app that natively integrates network, endpoint and cloud data to stop sophisticated attacks.",
     "support": "xsoar",
-    "currentVersion": "1.0.1",
+    "currentVersion": "1.1.0",
```
Comment (Itay4) [inline Packs/CortexXDR/pack_metadata.json:5]:
```suggestion
    "currentVersion": "1.0.2",
```

File: Packs/CortexXDR/TestPlaybooks/playbook-Cortex_XDR.yml [hunk 2]
```diff
@@ -1277,131 +1298,235 @@ tasks:
...
         }
       }
+  "38":
+    conditions:
+    - condition:
+      - - left:
+            iscontext: true
+            value:
+              simple: PaloAltoNetworksXDR.Endpoint
+          operator: hasLength
```
Comment (Itay4) [inline Packs/CortexXDR/TestPlaybooks/playbook-Cortex_XDR.yml:1488]: i would use `greaterThanOrEqual` operator here for the test to be more stable

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py [hunk 3]
```diff
@@ -263,6 +263,9 @@ def get_endpoints(self,
                 json_data={}
             )
             endpoints = reply.get('reply')[search_from:search_to]
+            for endpoint in endpoints:
+                if not endpoint.get('endpoint_id'):
+                    endpoint['endpoint_id'] = endpoint.get('agent_id')
```
Comment (Itay4) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:268]: how do we know `agent_id` and `endpoint_id` are the same ?

Comment (ShahafBenYakir) [reply Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:268]: Playground 2020-06-04 10-42-20
Playground 2020-06-04 10-43-19
 **It is the same command - was returning once `agent_id` once `endpoint_id`**

**Comment by kirbles19** [comment]
@ShahafBenYakir Done.

**Pull request merged**